spec/services/cfe_civil/components/vehicles_spec.rb

```diff
@@ -33,5 +34,46 @@
         ],
       }.to_json)
     end
+
+    context "and when a partner vehicle has been added" do
+      before { create(:vehicle, :owned_by_partner, legal_aid_application:) }
+
+      it "returns the expected JSON block with only the applicant vehicle" do
+        expect(call).to eq({
+          vehicles: [
+            value: 2345.0,
+            loan_amount_outstanding: 321.0,
+            date_of_purchase: "2020-05-18",
+            in_regular_use: true,
+          ],
+        }.to_json)
+      end
+    end
+  end
+
+  context "when the partner data is requested and both client and partner have vehicles" do
+    subject(:call) { described_class.call(legal_aid_application, "partner") }
+
+    before do
+      create(:vehicle,
+             legal_aid_application:,
+             estimated_value: 2345.0,
+             payment_remaining: 321.0,
+             purchased_on: Date.new(2020, 5, 18),
+             used_regularly: true,
+             owner: "partner")
+      create(:vehicle, :populated, legal_aid_application:)
```

Comment:
i prefer the explicit value setting and test thereof here to the factory (i..e hidden values) method employed above at line 39. I would change the values to be different from those from the factory too to differentiate.